ENH: Add vtkMRMLStorageNode CompressionParameter and presets

Sunderlandkyl opened the pull request:
-Adds a framework for specifying different preset levels of compression in vtkMRMLStorageNode
-Changes in vtkStreamingVolumeCodec also allow codecs to specify their own presets
-Replaces previous CompressionLevel int

![image](https://user-images.githubusercontent.com/9222709/47896281-203f6200-de43-11e8-86ba-3fefd21fc138.png)

pieper commented:
Generally this looks great to me (I didn't have a chance to test or read too carefully).

One thing that occurs to me is that the "Show options" checkbox persists during a Slicer session, but is reset on restart.  

If we persist the state to settings, then users like Murat would be able to have easy access to these options for their large datasets (where picking a particular mode on a per-dataset level may save several minutes per save operation).

Sunderlandkyl committed:
ENH: Add vtkMRMLStorageNode CompressionParameter and presets

-Adds a framework for specifying different preset levels of compression in vtkMRMLStorageNode
-Changes in vtkStreamingVolumeCodec also allow codecs to specify their own presets
-Replaces previous CompressionLevel int

Sunderlandkyl commented:
The checkbox should now persist between sessions.

Sunderlandkyl commented:
@lassoan

lassoan commented:
Thank you! Merged - https://github.com/Slicer/Slicer/commit/900bc41a33cb099fef202a5edc28932e4da69d80.

I've changed some method names, so you may need to update extensions that use this.
